subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/maven/MavenBomResolveIntegrationTest.groovy

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve.maven
+
+import org.gradle.integtests.fixtures.AbstractHttpDependencyResolutionTest
+import org.gradle.integtests.fixtures.resolve.ResolveTestFixture
+
+class MavenBomResolveIntegrationTest extends AbstractHttpDependencyResolutionTest {
+    def resolve = new ResolveTestFixture(buildFile)
+
+    def setup() {
+        resolve.prepare()
+        settingsFile << """
+            rootProject.name = 'testproject'
+        """
+        buildFile << """
+            repositories { maven { url "${mavenHttpRepo.uri}" } }
+            configurations { compile }
+        """
+    }
+
+    def "interprets dependencies declared in BOMs of transitive dependencies as optional dependencies"() {
```

Comment:
In practice this doesn't happen yet, because no one "depends" on a BOM in Maven land, you "import" it into your own depdencyManagement block.

I think the more important test case is when the Gradle project has a first level dependency on a BOM, which is the way we want to model "BOM import" in Gradle.